app/server/lib/AttachmentFileManager.ts

```diff
@@ -175,38 +175,41 @@ export class AttachmentFileManager extends EventEmitter {
     if (!fileMetadata) { return false; }
     if (await this._isFileAvailable(fileMetadata)) { return false; }
 
-    const originalStoreExists = fileMetadata.ident in this._getStoreProvider().listAllStoreIds();
-    const newStoreId = originalStoreExists ? fileMetadata.ident : defaultStoreId;
-    const newStore = newStoreId && await this._getStore(newStoreId) || undefined;
-
-    if (newStoreId && !newStore) {
-      throw new StoreNotAvailableError(newStoreId);
+    // Try to use the store the file was originally uploaded to, if it's available.
+    const originalStoreId = fileMetadata.storageId;
+    const originalStoreStillExists = !!originalStoreId && originalStoreId in this._getStoreProvider().listAllStoreIds();
+    const destinationStoreId = originalStoreStillExists ? originalStoreId : defaultStoreId;
```

Comment:
This is the bugfix, right? There might not have been an original store ID at all.

Reply:
Bugfix was it was originally using `fileMetadata.ident`, and it should have been using `fileMetadata.storageId` - it was looking up the file's unique identifier in the list of stores :facepalm: